controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -222,11 +222,22 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 			reterr = kerrors.NewAggregate([]error{reterr, err})
 		}
 
-		// TODO: remove this as soon as we have a proper remote cluster cache in place.
-		// Make KCP to requeue in case status is not ready, so we can check for node status without waiting for a full resync (by default 10 minutes).
-		// Only requeue if we are not going in exponential backoff due to error, or if we are not already re-queueing, or if the object has a deletion timestamp.
-		if reterr == nil && !res.Requeue && res.RequeueAfter <= 0 && kcp.ObjectMeta.DeletionTimestamp.IsZero() {
-			if !kcp.Status.Ready {
+		if reterr == nil && kcp.ObjectMeta.DeletionTimestamp.IsZero() {
+			// TODO: remove this as soon as we have a proper remote cluster cache in place.
+			// Make KCP to requeue in case status is not ready, so we can check for node status without waiting for a full resync (by default 10 minutes).
+			// Only requeue if we are not going in exponential backoff due to error, or if we are not already re-queueing, or if the object has a deletion timestamp.
+			if !res.Requeue && res.RequeueAfter <= 0 {
```

Comment:
If I understand right we could fix this by having KCP watch nodes in the same way the the MachineSet controller does. It's probably not necessary to make this change now - but maybe we should update the comment here to make that clear.

Reply:
it's the below case which is new. I noted:

```
			// The alternative solution to this requeue would be watching the relevant pods inside each workload
			// cluster which would be very expensive.
```

But happy to add better wording there 🙂

Reply:
I mean this for the older comment - will add a suggestion.

Reply:
We would have to watch pods not nodes which I don't like: https://github.com/kubernetes-sigs/cluster-api/issues/8786#issuecomment-1645474641